src/runtime/typemanager.cs

```diff
@@ -76,11 +77,22 @@ internal static void RestoreRuntimeData(RuntimeDataStorage storage)
         {
             Debug.Assert(cache == null || cache.Count == 0);
             storage.GetValue("slots", out _slotsImpls);
-            storage.GetValue("cache", out cache);
-            foreach (var entry in cache)
+            var _cache = new Dictionary<MaybeType, IntPtr>();
+            storage.GetValue("cache", out _cache);
+            foreach (var entry in _cache)
             {
-                Type type = entry.Key;
+                Type type = null;
+                try
+                {
+                    type = entry.Key.Value;
+                }
+                catch
+                {
+                    Runtime.XDecref(entry.Value);
+                    continue;
+                }
```

Comment:
Is there a specific exception we could handle?
Can the exception be avoided in the first place by adding a precheck?

Reply:
> ... adding a precheck?

Yes